File: app/src/test/java/info/nightscout/androidaps/plugins/general/automation/elements/InputTempTargetTest.java
package info.nightscout.androidaps.plugins.general.automation.elements;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.powermock.core.classloader.annotations.PrepareForTest;
import org.powermock.modules.junit4.PowerMockRunner;

import info.AAPSMocker;
import info.nightscout.androidaps.Constants;
import info.nightscout.androidaps.MainApp;
import info.nightscout.androidaps.plugins.configBuilder.ProfileFunctions;

@RunWith(PowerMockRunner.class)
@PrepareForTest({MainApp.class, ProfileFunctions.class})
public class InputTempTargetTest {

    @Test
    public void getSetValueTest() {
        InputTempTarget i = new InputTempTarget().setUnits(Constants.MMOL).setValue(5d);
        Assert.assertEquals(5d, i.getValue(), 0.01d);
        Assert.assertEquals(Constants.MIN_TT_MMOL, i.minValue, 0.01d);
        i = new InputTempTarget().setValue(100d).setUnits(Constants.MGDL);
        Assert.assertEquals(100d, i.getValue(), 0.01d);
        Assert.assertEquals(Constants.MIN_TT_MGDL, i.minValue, 0.01d);
        Assert.assertEquals(Constants.MGDL, i.getUnits());
    }


    @Before
    public void prepare() {
        AAPSMocker.mockMainApp();
        AAPSMocker.mockStrings();
        AAPSMocker.mockProfileFunctions();
    }
}

File: automation/src/test/java/info/nightscout/androidaps/plugins/general/automation/elements/InputTempTargetTest.kt
package info.nightscout.androidaps.plugins.general.automation.elements

import info.nightscout.androidaps.interfaces.GlucoseUnit
import info.nightscout.androidaps.plugins.general.automation.triggers.TriggerTestBase
import org.junit.Assert
import org.junit.Test

class InputTempTargetTest : TriggerTestBase() {

    @Test fun setValueTest() {
        val i = InputTempTarget(profileFunction)
        i.units = GlucoseUnit.MMOL
        i.value = 5.0
        Assert.assertEquals(5.0, i.value, 0.01)
        i.units = GlucoseUnit.MGDL
        i.value = 100.0
        Assert.assertEquals(100.0, i.value, 0.01)
        Assert.assertEquals(GlucoseUnit.MGDL, i.units)
    }
}